internal/reuse/reuse.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: Copyright 2025 SAP SE
+// SPDX-License-Identifier: Apache-2.0
+
+package reuse
+
+import (
+	"embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/sapcc/go-bits/must"
+
+	"github.com/sapcc/go-makefile-maker/internal/core"
+	"github.com/sapcc/go-makefile-maker/internal/golang"
+)
+
+//go:embed go-licence-detector.tmpl
+var goLicenceDetectorTemplate embed.FS
+
+var reuseTemplate = strings.ReplaceAll(`# SPDX-FileCopyrightText: SAP SE
+# SPDX-License-Identifier: Apache-2.0
+version = 1
+
+[[annotations]]
+path = [
+	".github/CODEOWNERS",
+	".github/renovate.json",
+	".gitignore",
+	".license-scan-overrides.jsonl",
+	".license-scan-rules.json",
+	"go.mod",
+	"go.sum",
+	"Makefile.maker.yaml",
+	"vendor/modules.txt",
+]
+SPDX-FileCopyrightText = "SAP SE"
+SPDX-License-Identifier = "Apache-2.0"
+%[1]s`, "\t", "  ")
+
+func RenderConfig(cfg core.Configuration, sr golang.ScanResult) {
+	// If disabled, the REUSE.toml file should not be overridden.
+	// This is useful if the project needs additional information in
+	// the REUSE.toml file, e.g., specific disclaimers.
+	if cfg.Reuse.Enabled != nil && !*cfg.Reuse.Enabled {
+		return
+	}
+
+	gLDT := must.Return(goLicenceDetectorTemplate.Open("go-licence-detector.tmpl"))
+	defer gLDT.Close()
+
+	tmpGLDT := must.Return(os.CreateTemp("", "go-makefile-maker-*"))
+	defer os.Remove(tmpGLDT.Name())
+
+	_ = must.Return(io.Copy(tmpGLDT, gLDT))
+
+	_ = must.Return(tmpGLDT.Seek(0, 0))
+
+	cmd := exec.Command("bash", "-c", //nolint:gosec // only a temporary file the user self inflicts
+		"go list -m -mod=readonly -json all | go-licence-detector -includeIndirect -rules .license-scan-rules.json -overrides .license-scan-overrides.jsonl -depsOut /dev/stdout -depsTemplate "+"/dev/fd/"+strconv.Itoa(int(tmpGLDT.Fd())))
```

Comment:
The fd number of the current process is not indicative of what the child process will see. ExtraFiles is specified like this: "If non-nil, entry i becomes file descriptor 3+i." So `ExtraFiles[0]` becomes FD 3 in the child process always.
```suggestion
	cmd := exec.Command("bash", "-c", //nolint:gosec // only a temporary file the user self inflicts
		"go list -m -mod=readonly -json all | go-licence-detector -includeIndirect -rules .license-scan-rules.json -overrides .license-scan-overrides.jsonl -depsOut /dev/stdout -depsTemplate /dev/fd/3")
```